roles/gitian/templates/gitian-build.sh

```diff
@@ -172,34 +180,58 @@ fi
 echo ${COMMIT}
 
 # Set up build
-pushd ./zcash
+pushd ${zcash_repo_dir_path}
 git fetch
 git checkout ${COMMIT}
 popd
 
+explode_yaml_file.py ${zcash_repo_dir_path}/contrib/gitian-descriptors/gitian-linux.yml suites ${suite_descriptors_dir_path}
+suites=$(ls ${suite_descriptors_dir_path})
```

Comment:
This line was an earlier version that got a list of suites by listing files in a directory. The current version gets that list instead from the output of the python script, so that the content of the descriptor file can be used to control the order in which suites are built.

I don't think that addresses the concern about allowable characters, but as you indicated, probably fine in practice.